internal/provider/resource_saml_group_mapping.go

```diff
@@ -0,0 +1,375 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"slices"
+	"strings"
+
+	"github.com/google/uuid"
+	"wiz.io/hashicorp/terraform-provider-wiz/internal/utils"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"wiz.io/hashicorp/terraform-provider-wiz/internal"
+	"wiz.io/hashicorp/terraform-provider-wiz/internal/client"
+	"wiz.io/hashicorp/terraform-provider-wiz/internal/wiz"
+)
+
+// ReadSAMLGroupMappings represents the structure of a SAML group mappings read operation.
+// It includes a SAMLIdentityProviderGroupMappingsConnection object.
+type ReadSAMLGroupMappings struct {
+	SAMLGroupMappings wiz.SAMLIdentityProviderGroupMappingsConnection `json:"samlIdentityProviderGroupMappings"`
+}
+
+// SAMLGroupMappingsImport represents the structure of a SAML group mapping import.
+// It includes the SAML IdP ID, provider group ID, project IDs, and role.
+type SAMLGroupMappingsImport struct {
+	SamlIdpID       string
+	ProviderGroupID string
+	ProjectIDs      []string
+	Role            string
+}
+
+// UpdateSAMLGroupMappingPayload struct
+type UpdateSAMLGroupMappingPayload struct {
+	SAMLGroupMapping wiz.SAMLGroupMapping `json:"samlGroupMapping,omitempty"`
+}
+
+func resourceWizSAMLGroupMapping() *schema.Resource {
+	return &schema.Resource{
+		Description: "Configure SAML Group Role Mapping. When using SSO to authenticate with Wiz, you can map group memberships in SAML assertions to Wiz roles across specific scopes.",
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Description: "Unique tf-internal identifier for the saml group mapping",
+				Computed:    true,
+			},
+			"saml_idp_id": {
+				Type:        schema.TypeString,
+				Description: "Identifier for the Saml Provider",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"provider_group_id": {
+				Type:        schema.TypeString,
+				Description: "Provider group ID",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"role": {
+				Type:        schema.TypeString,
+				Description: "Wiz Role name",
+				Required:    true,
+			},
+			"projects": {
+				Type:        schema.TypeList,
+				Optional:    true,
```

Comment:
Was adding this as a block set, containing the mappings of provider_group_id, role and projects considered? This way it could be a dynamic block in the same resource as opposed to a resource per saml idp mapping

Reply:
So that the terraform resource in the end would look something like this?
```
resource "wiz_saml_group_mapping" "saml_mapping_test" {
  saml_idp_id = "wiz-azure-ad-saml"
  mappings = [
      provider_group_id = "something"
      role = "PROJECT_READER"
      projects = [
           "some_id",
           "some_id"
      ]
  ]
}
```

Reply:
Something like:
```hcl
resource "wiz_saml_group_mapping" "saml_mapping_test" {
  saml_idp_id = "wiz-azure-ad-saml"
  mapping{
    provider_group_id = "global_grp1"
    role = "GLOBAL_ADMIN"
  }
  mapping{
    provider_group_id = "proj_member"
    role = "PROJECT_MEMBER"
    projects = ["hexid1, "hexid2"]
}
}
```

Much like the existing saml_idp resource, you can reference that.

Reply:
I would go for this:
```
resource "wiz_saml_group_mapping" "saml_mapping_test" {
  saml_idp_id = "wiz-azure-ad-saml"
  mappings = [
    {
      provider_group_id = "global_grp1"
      role = "GLOBAL_ADMIN"
    },
    {
      provider_group_id = "proj_member"
      role = "PROJECT_MEMBER"
      projects = ["hexid1", "hexid2"]
    },
  ]
}
```

Reply:
State drift will become a challenge with that approach, if any part of the list changes outside of Terraform, the drift detection would mark everything as a speculative change, making it harder to identify the specific element that caused the drift. Generally, if there are going to be a large number of mappings, it is going to be very difficult to review speculative plans if there is for example a small update to a single mapping.

When implementing this provider, [dynamic blocks](https://developer.hashicorp.com/terraform/language/expressions/dynamic-blocks) are typically used to build multiple blocks of `mappings` for example